src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs

```diff
@@ -277,8 +277,17 @@ public async Task InitializeAsync(IEnumerable<FunctionMetadata> functions, Cance
 
             var workerConfig = _workerConfigs.Where(c => c.Description.Language.Equals(_workerRuntime, StringComparison.InvariantCultureIgnoreCase)).FirstOrDefault();
 
+            // For other OOP workers, workerconfigs are present inside "workers" folder of host bin directory and is used to populate "_workerConfigs".
+            // For dotnet-isolated _workerConfigs is populated by reading workerconfig.json from the deployed payload(customer function app code).
+            // So if workerConfig is null and worker runtime is dotnet-isolated, that means isolated function code was not deployed yet.
+            var isDotNetIsolatedAppWithoutPayload = string.Equals(_workerRuntime, RpcWorkerConstants.DotNetIsolatedLanguageWorkerName, StringComparison.InvariantCultureIgnoreCase)
+                                                    && workerConfig == null;
+
             // We are skipping this check for multi-language environments because they use multiple workers and thus doesn't honor 'FUNCTIONS_WORKER_RUNTIME'
-            if ((workerConfig == null && (functions == null || functions.Count() == 0)) && !_environment.IsMultiLanguageRuntimeEnvironment())
+            // Also, skip if dotnet-isoalted app without payload as it is a valid case to exist.
+            if ((workerConfig == null && (functions == null || functions.Count() == 0))
+                && !_environment.IsMultiLanguageRuntimeEnvironment()
+                && !isDotNetIsolatedAppWithoutPayload)
```

Comment:
Do we want to add any additional logs here?

Reply:
I felt like it was not needed as it is a valid use case to go through. Can add if folks think this information might be useful. I felt that the other information we log is enough to derive this data.